### DISCUSSION TITLE
Does anyone know how to have syntax highlighting on svelte files?

### DISCUSSION THREAD

**Discussion by Sophistiqq** [NvChad/NvChad#2894] [Q&A] [answered]
`nvim-treesitter[svelte]: Error during "tree-sitter generate"
node:internal/modules/cjs/loader:1148
  throw err;
  ^

Error: Cannot find module 'tree-sitter-html/grammar'
Require stack:
- /home/roi/.local/share/nvim/tree-sitter-svelte/grammar.js
- /home/roi/.local/share/nvim/tree-sitter-svelte/[stdin]
    at Module._resolveFilename (node:internal/modules/cjs/loader:1145:15)
    at Module._load (node:internal/modules/cjs/loader:986:27)
    at Module.require (node:internal/modules/cjs/loader:1233:19)
    at require (node:internal/modules/helpers:179:18)
    at Object.<anonymous> (/home/roi/.local/share/nvim/tree-sitter-svelte/grammar.js:11:14)
    at Module._compile (node:internal/modules/cjs/loader:1358:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1416:10)
    at Module.load (node:internal/modules/cjs/loader:1208:32)
    at Module._load (node:internal/modules/cjs/loader:1024:12)
    at Module.require (node:internal/modules/cjs/loader:1233:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/roi/.local/share/nvim/tree-sitter-svelte/grammar.js',
    '/home/roi/.local/share/nvim/tree-sitter-svelte/[stdin]'
  ]
}

Node.js v20.14.0
Failed to load grammar.js

Caused by:
    node process exited with status 1

Press ENTER or type command to continue`

**Comment by Sophistiqq** [comment]
I know this is a bit complicated, so I just set the filetype to html and it's giving me the highlighting, however, I want to set the default filetype of svelte files to html.

**Comment by siduck** [comment]
> filetype of svelte files to html.
thats bad,why do that?

**Reply by Sophistiqq** [reply, marked as answer]
> > filetype of svelte files to html.
> > thats bad,why do that?

Treesitter on my machine is kinda neglecting the existence of svelte files, However setting filetype of svelte files to html works, I can have the syntax highlighting when I do :setfiletype html on svelte files and it's okay for me.

I just need now how can I automatically do it without typing the commands everytime.

EDIT

Found a simple solution now, I just created another table entry anywhere,
{
    vim.filetype.add({
      extension = {
        svelte = "html",
      },
    })
 },

**Reply by siduck** [reply]
@Sophistiqq  TSInstall svelte?

**Reply by Sophistiqq** [reply]
> @Sophistiqq TSInstall svelte?

doesn't work too, even installing TSInstallFromGrammar, it will install, yes, but it's not giving syntax highlighting, html's highlighting works enough, also lsp is working fine.

**Reply by siduck** [reply]
working fine here

![image](https://github.com/NvChad/NvChad/assets/59060246/23d8a221-8258-400a-82b3-0c8da7016be5)


Ig the problem is your system :joy:
